src/rbd.cc

```diff
@@ -771,23 +771,44 @@ static int do_purge_snaps(librbd::Image& image)
 {
   MyProgressContext pc("Removing all snapshots");
   std::vector<librbd::snap_info_t> snaps;
+  bool is_protected = false;
   int r = image.snap_list(snaps);
   if (r < 0) {
     pc.fail();
     return r;
   }
+   
+  else if (0 == snaps.size())
+  {
+	return 0;
+  }
```

Comment:
sorry for my verbose, the style should be:
if (r < 0) {
     pc.fail();
     return r;
   } else if (0 == snaps.size()) {
+	return 0;
+  } else {
+ .....